### PR TITLE
Save state

### PR DESCRIPTION
**Pull request by khalp**
Added get started/welcome screen and fixed other bugs related to orientation/spanning changes

Note: still haven't fixed bug where rotating during selection mode erases all selections

Other note: turns out I actually had pushed a gradle version change in my last PR so I reset it here (reverting it seemed to fix my "default activity not found" problems!)

### CODE REVIEW

File: TwoNote/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -18,4 +18,7 @@
     <string name="selected">selected</string>
     <string name="load_error_note">Error: loaded file is not of type Note</string>
     <string name="load_error_dir">Error: loaded file is not of type DirEntry</string>
+    <string name="welcome">Welcome to Two Note</string>
```
Comment (parkerschroeder) [inline TwoNote/app/src/main/res/values/strings.xml:21]: we never talked about this, but should our app be named "TwoNote" or "Two Note"?

File: TwoNote/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Tue Jul 21 08:59:00 EDT 2020
+#Thu Jul 23 14:36:48 EDT 2020
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.1.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-all.zip
```
Comment (parkerschroeder) [inline TwoNote/gradle/wrapper/gradle-wrapper.properties:6]: is this what fixed your MainActivity issue?

File: TwoNote/app/src/main/res/values/strings.xml (hunk 1 above)
Comment (khalp) [reply TwoNote/app/src/main/res/values/strings.xml:21]: that is a good question, if we're following OneNote then I guess TwoNote? I don't mind either way

File: TwoNote/gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (khalp) [reply TwoNote/gradle/wrapper/gradle-wrapper.properties:6]: yeah this and clicking File > Invalidate cache/restart

File: TwoNote/app/src/main/res/layout/fragment_get_started.xml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.widget.LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (parkerschroeder) [inline TwoNote/app/src/main/res/layout/fragment_get_started.xml:2]: The getting started fragment looks fantastic

File: TwoNote/app/src/main/java/com/microsoft/device/display/samples/twonote/NoteSelectionListener.kt [hunk 4]
```diff
@@ -6,32 +6,43 @@ import android.view.MenuItem
...
                     }
-                    arrayAdapter?.notifyDataSetChanged()
                 }
+                arrayAdapter.notifyDataSetChanged()
                 onDestroyActionMode(mode)
+
+                // If spanned, check if the displayed NoteDetailFragment has been deleted
+                if (ScreenHelper.isDualMode(host.requireActivity())) {
+                    // REVISIT: couldn't get findFragmentById to work but this needs to be changed to something more efficient eventually
+                    val inode = host.parentFragmentManager.fragments.find { frag -> frag is NoteDetailFragment }?.arguments?.getSerializable(MainActivity.INODE) as? INode
```
Comment (parkerschroeder) [inline TwoNote/app/src/main/java/com/microsoft/device/display/samples/twonote/NoteSelectionListener.kt:38]: consider splitting this into two so the line isn't so long

File: TwoNote/app/src/main/java/com/microsoft/device/display/samples/twonote/NoteDetailFragment.kt [hunk 5]
```diff
@@ -275,20 +265,46 @@ class NoteDetailFragment : Fragment() {
...
+        activity?.let {
+            if (ScreenHelper.isDualMode(it)) {
+                parentFragmentManager.beginTransaction()
+                    .replace(R.id.dual_screen_end_container_id, GetStartedFragment(), null)
+                    .commit()
+            } else {
+                parentFragmentManager.beginTransaction()
+                    .replace(R.id.single_screen_container_id, NoteListFragment(), null)
+                    .commit()
+                // it.onBackPressed()
```
Comment (parkerschroeder) [inline TwoNote/app/src/main/java/com/microsoft/device/display/samples/twonote/NoteDetailFragment.kt:307]: ooh, looks like some potential infinite recursion. you should uncomment this and see what happens

Comment (khalp) [reply TwoNote/app/src/main/java/com/microsoft/device/display/samples/twonote/NoteDetailFragment.kt:307]: haha it actually refers to the activity's onBackPressed (which is related to the back gesture!)

**Pull request merged**